.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,46 @@
+name: Doxygen
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  CI_Doxygen:
+    name: ubuntu_debug_x64
+    runs-on: ubuntu-latest
+    container:
+      image: novelrt/novelrt-build:latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Run CI Build Script (Engine & Docs only)
+        run: scripts/cibuild.sh --configuration Debug --generate -DNOVELRT_BUILD_TESTS OFF -DNOVELRT_BUILD_SAMPLES OFF
+
+      - name: Generate Documentation
+        uses: mattnotmitt/doxygen-action@v1
+        id: generate
+        with:
+            working-directory: '$GITHUB_WORKSPACE/novelrt/artifacts/build/Debug'
+            doxyfile-path: 'doxygen/Doxyfile.doxygen'
+
+      - name: Copy Documentation
+        id: copy
+        if: ${{ success() }}
+        run: mkdir -p $GITHUB_WORKSPACE/novelrt/docs && cp -r -f -v $GITHUB_WORKSPACE/novelrt/artifacts/build/Debug/doxygen/html/* $GITHUB_WORKSPACE/novelrt/docs
```

Comment:
I believe we can set `DOXYGEN_OUTPUT_DIRECTORY` as a CMake variable to `artifacts/docs` or some such (as part of cibuild, most likely) to avoid this step entirely.

Reply:
Yup! I wrote another comment wondering about that CMake var. Resolved.